brut.apktool/apktool-lib/src/main/java/brut/androlib/src/SmaliDecoder.java

```diff
@@ -92,21 +103,134 @@ private DexFile decode() throws AndrolibException {
             }
 
             if (dexFile instanceof DexBackedOdexFile) {
-                options.inlineResolver =
+                config.inlineResolver =
                         InlineMethodResolver.createInlineMethodResolver(((DexBackedOdexFile)dexFile).getOdexVersion());
             }
 
-            Baksmali.disassembleDexFile(dexFile, mOutDir, jobs, options);
+            if (config.resolveResources)
+                LOGGER.info("Parsing resource ids in " + config.dexFile + "...");
+            disassembleDexFile(dexFile, config.outDir, jobs, config);
 
             return dexFile;
         } catch (IOException ex) {
             throw new AndrolibException(ex);
         }
     }
 
-    private final File mApkFile;
-    private final File mOutDir;
-    private final String mDexFile;
-    private final boolean mBakDeb;
-    private final int mApiLevel;
+    public static boolean disassembleDexFile(DexFile dexFile, File outputDir, int jobs, final Config config) {
+        return disassembleDexFile(dexFile, outputDir, jobs, config, null);
+    }
+
+    public static boolean disassembleDexFile(DexFile dexFile, File outputDir, int jobs, final Config config,
```

Comment:
Is this where the magic happens? We override the disassembly step to override the `call` function? Just struggling at present with knowing what is a port from smali and where our custom patch comes in. Or is it fair to say all these new lines are additions from you?

Reply:
This was ported over directly from baksmali to change the parameters of disassembleDexFile to now accept the Config. The magic happens inside of ResourceDecoder and ApkDecoder.